op-e2e/actions/upgrades/isthmus_fork_test.go

```diff
@@ -601,3 +603,162 @@ func TestSetCodeTxTypeIsthmus(gt *testing.T) {
 	require.Equal(t, verifier.L2Unsafe(), verifier.L2Safe())
 	require.Equal(t, sequencer.L2Safe(), verifier.L2Safe())
 }
+
+func TestSetCodeTxTypePreIsthmus(gt *testing.T) {
+	// Ensure that batches that include SetCodeTxs are dropped if before Isthmus
+	// Sets up a network with Isthmus starting at block 1
+	// Send SetCodeTx at block 3
+	// Verifier pipeline uses Isthmus starting at block 5
+	// Ensure verifier drops the batch with a SetCodeTx too early
+
+	t := helpers.NewDefaultTesting(gt)
+	p := &e2eutils.TestParams{
+		MaxSequencerDrift:   20,
+		SequencerWindowSize: 24,
+		ChannelTimeout:      20,
+		L1BlockTime:         12,
+		AllocType:           config.AllocTypeStandard,
+	}
+	dp := e2eutils.MakeDeployParams(t, p)
+	dp.DeployConfig.ActivateForkAtOffset(rollup.Isthmus, 2)
+	minTs := hexutil.Uint64(0)
+	upgradesHelpers.ApplyDeltaTimeOffset(dp, &minTs)
+
+	var (
+		aa = common.HexToAddress("0x000000000000000000000000000000000000aaaa")
+		bb = common.HexToAddress("0x000000000000000000000000000000000000bbbb")
+	)
+
+	// Create 2 contracts, (1) writes 42 to slot 42, (2) calls (1)
+	store42Program := program.New().Sstore(0x42, 0x42)
+	callBobProgram := program.New().Call(nil, dp.Addresses.Bob, 1, 0, 0, 0, 0)
+
+	alloc := helpers.DefaultAlloc
+	alloc.L2Alloc = make(map[common.Address]types.Account)
+	alloc.L2Alloc[aa] = types.Account{
+		Code: store42Program.Bytes(),
+	}
+	alloc.L2Alloc[bb] = types.Account{
+		Code: callBobProgram.Bytes(),
+	}
```

Comment:
Do we really need to set this all up and have a non-trivial SetCode tx? If this test is mostly about testing that derivation drops SetCode txs, you can just create one with empty auth list.